DNS resolution control

murrant opened the pull request:
Fixes hosts file lookups
clears cached IP when NXDOMAIN is received (but not SERVERFAIL)
Adds `dns.resolution_mode` setting to control DNS resolution.  This should pretty much always be the default "os" on modern OS.

Mostly fixing issues with #18537 

fixes #18537

Note: ext-sockets was already required, just made it explicit.

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

laf commented on resources/definitions/config_definitions.json:
```diff
@@ -1942,6 +1942,20 @@
             "type": "integer"
         },
+        "dns.resolution_mode": {
+            "default": "os_default",
```
`os`?

murrant replied:
yes, I renamed the options and I missed this :)

murrant commented:
ok so for example, on the firewall, you block snmp on ipv6, but allow on ipv4?

Dang, that case sucks :)

Jellyfrog commented:
Honestly this seems insane? 😆 Why not just leave the DNS caching and resolving to the OS?

murrant commented:
Also, a little more history.
1. I tried to use our cached resolved DNS #18527 that had bugs because:
   a. it didn't pull in the hosts file
   b. dns could return AAAA record even if IPv6 is not available/connectable
2. I switched to socket_addrinfo_lookup(), which uses getaddrinfo.  This easily fixed the hosts file problem, but did fix the connectivity problem.
3. Likely, we want to check connectivity to determine the correct address.  Happy Eyeballs does this.  Discovered that only supports connection oriented protocols
4. RFC6555 is pretty simple, give it a shot to implement to learn a bit about fibers.
5. Working implementation, discovered we have a reactphp dependency already and it has an RFC6555 implementation.  (which of course only supports TCP)
6. Build test ReactPHP UDP implementation making it easy to share udp protocol verification across implementations.
7. we are here :D decide what to do with this code:
    a. throw it away, keep many dns queries during polling and incorrect IP display on the device page in some cases.
    b. extract some of socket_addrinfo_lookup() code to display a more correct IP or maybe a list of IPs.
    c. pick an implementation and move forward with this allowing full control of how DNS is resolved within the application and accurate device IP.  More code to maintain, that probably has some bugs. 

So yeah, it came full circle.  I started off with something simple (which I didn't expect to get merged) and tried something complex after corner cases were highlighted.  We reverted out the original change.
